ui/src/components/SampleView/VideoPlayer.jsx

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useRef } from 'react';
+import { useSelector } from 'react-redux';
+
+import { JSMpeg } from './jsmpeg.min.js';
+import styles from './VideoPlayer.module.css';
+
+export default function VideoPlayer() {
+  const videoWidth = useSelector((state) => state.sampleview.width);
+  const videoHeight = useSelector((state) => state.sampleview.height);
+  const videoFormat = useSelector((state) => state.sampleview.videoFormat);
+  const videoURL = useSelector((state) => state.sampleview.videoURL);
+  const videoHash = useSelector((state) => state.sampleview.videoHash);
+
+  const videoSource = `${videoURL}/${videoHash}`;
+
+  const canvasRef = useRef(null);
+  const playerRef = useRef(null);
+
+  useEffect(() => {
+    const canvas = canvasRef.current;
+    if (!canvas || videoFormat !== 'MPEG1') {
+      return () => {};
+    }
+
+    canvas.width = videoWidth;
+    canvas.height = videoHeight;
+
+    if (playerRef.current) {
+      playerRef.current.destroy();
+      playerRef.current = null;
+    }
+
+    playerRef.current = new JSMpeg.Player(videoSource, {
+      canvas,
+      decodeFirstFrame: false,
+      preserveDrawingBuffer: true,
+      protocols: [],
+    });
+
+    playerRef.current.play();
+
+    return () => {
+      if (playerRef.current) {
+        playerRef.current.destroy();
+        playerRef.current = null;
+      }
+    };
+  }, [videoFormat, videoSource, videoWidth, videoHeight]);
+
+  if (!videoURL) {
+    return null;
+  }
+
+  switch (videoFormat) {
```

Comment:
I tend to prefer multiple consecutive `if () { return }` blocks to remove one level of indentation.